.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -0,0 +1,428 @@
+  let allIssues = [];
+  try {
+    while (true) {
+      const response = await github.issues.listForRepo(queryParams);
+      // filter out pull requests
+      const issues = filterPrs ? response.data.filter(issue => !issue.pull_request) : response.data;
+      allIssues = allIssues.concat(issues);
+
+      // Check the 'link' header to see if there are more pages
+      const linkHeader = response.headers.link;
+      if (!linkHeader || !linkHeader.includes('rel="next"')) {
+        break;
+      }
+
+      queryParams.page++;
+    }
+    return allIssues;
+  } catch (error) {
+    console.error('Error fetching issues:', error);
+    return [];
+  }
+}
+
+function genLookbackDates() {
+  const now = new Date()
+  const midnightYesterday = new Date(
+    Date.UTC(
+      now.getUTCFullYear(),
+      now.getUTCMonth(),
+      now.getUTCDate(),
+      0, 0, 0, 0
+    )
+  );
+  const sevenDaysAgo = new Date(midnightYesterday);
+  sevenDaysAgo.setDate(midnightYesterday.getDate() - 7);
+  return { sevenDaysAgo, midnightYesterday };
+}
+
+function filterOnDateRange({ issue, sevenDaysAgo, midnightYesterday }) {
+  const createdAt = new Date(issue.created_at);
+  return createdAt >= sevenDaysAgo && createdAt <= midnightYesterday;
+}
+
+async function getNewIssues({github, context}) {
+  const { sevenDaysAgo, midnightYesterday } = genLookbackDates();
+  const queryParams = {
+    owner: REPO_OWNER,
+    repo: REPO_NAME,
+    state: 'all', // To get both open and closed issues
+    per_page: 100, // Number of items per page (maximum allowed)
+    page: 1, // Start with page 1
+    since: sevenDaysAgo.toISOString(),
+  };
+
+  try {
+    const allIssues = await getIssues(github, queryParams)
+    const filteredIssues = allIssues.filter(issue => filterOnDateRange({ issue, sevenDaysAgo, midnightYesterday }));
+    return filteredIssues;
+  } catch (error) {
+    console.error('Error fetching issues:', error);
+    return [];
+  }
+}
+
+async function getTargetLabelIssues({octokit, labels, filterPrs, context}) {
+  const queryParams = {
+    owner: REPO_OWNER,
+    repo: REPO_NAME,
+    state: 'open',
+    per_page: 100, // Number of items per page (maximum allowed)
+    page: 1, // Start with page 1
+    labels
+  };
+  debug({msg: "fetching issues", queryParams})
+  try {
+    const allIssues = await getIssues(octokit, queryParams, filterPrs)
+    return allIssues;
+  } catch (error) {
+    console.error('Error fetching issues:', error);
+    return [];
+  }
+}
+
+/**
+ * Get data required for issues report
+ */
+async function getIssuesData({github, context}) {
+  const targetLabels = {
+    "needs triage": {
+      filterPrs: true,
+      alias: "issuesTriage",
+    },
+    "ready to merge": {
+      filterPrs: false,
+      alias: "issuesReadyToMerge",
+    },
+    "Sponsor Needed": {
+      filterPrs: true,
+      alias: "issuesSponsorNeeded",
+    },
+  };
+
+  const issuesNew = await getNewIssues({github: github.rest, context});
```

Comment:
This is nitpicking, but hey names are important ;) The use of the `github` name is a bit frivolous. You rename the `github.rest` to `octokit` in other places - perhaps stick to that?

Reply:
updated